File: src/main/java/com/zakpruitt/pbst/entities/PokemonCard.java
package com.zakpruitt.pbst.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PokemonCard {

    @Id
    private String id;

    private String name;
    private String setCode;
    private String setName;
    private String cardNumber;
    private String rarity;
    private String imageUrl;

    private BigDecimal marketPrice;
    private BigDecimal lowPrice;

    private LocalDateTime lastPriceSync;
}

File: internal/models/pokemon_card.go
package models

import (
	"database/sql"
	"time"

	"gorm.io/gorm"
)

type PokemonCard struct {
	ID            string         `gorm:"primaryKey" json:"id"`
	Name          string         `json:"name"`
	SetCode       string         `gorm:"column:set_code" json:"set_code"`
	SetName       string         `gorm:"column:set_name" json:"set_name"`
	CardNumber    string         `gorm:"column:card_number" json:"card_number"`
	Rarity        string         `json:"rarity"`
	ImageURL      string         `gorm:"column:image_url" json:"image_url"`
	MarketPrice   float64        `gorm:"column:market_price" json:"market_price"`
	LowPrice      float64        `gorm:"column:low_price" json:"low_price"`
	LastPriceSync sql.NullTime   `gorm:"column:last_price_sync" json:"last_price_sync"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
}
